contracts/instruments/StakedPut.sol

```diff
@@ -0,0 +1,330 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.7.2;
+pragma experimental ABIEncoderV2;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {DSMath} from "../lib/DSMath.sol";
+import {
+    AggregatorV3Interface
+} from "@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol";
+
+import {
+    State,
+    IHegicOptions,
+    HegicOptionType,
+    IHegicETHOptions,
+    IHegicBTCOptions,
+    IHegicRewards
+} from "../interfaces/HegicInterface.sol";
+import {
+    ProtocolAdapterTypes,
+    IProtocolAdapter
+} from "../adapters/IProtocolAdapter.sol";
+import {IAmmAdapter} from "../adapters/IAmmAdapter.sol";
+import {AmmAdapter} from "../adapters/AmmAdapter.sol";
+
+import {ProtocolAdapter} from "../adapters/ProtocolAdapter.sol";
+import {IRibbonFactory} from "../interfaces/IRibbonFactory.sol";
+import {IWETH} from "../interfaces/IWETH.sol";
+import {IUniswapV2Pair} from "../interfaces/IUniswapV2Pair.sol";
+import {OtokenInterface} from "../interfaces/GammaInterface.sol";
+import {UniswapAdapter} from "../adapters/UniswapAdapter.sol";
+
+import {StakedPutStorageV1} from "../storage/StakedPutStorage.sol";
+
+contract StakedPut is DSMath, StakedPutStorageV1 {
+    event PositionCreated(
+        address indexed account,
+        uint256 indexed positionID,
+        uint256 amount
+    );
+    event Exercised(
+        address indexed account,
+        uint256 indexed positionID,
+        uint256 totalProfit
+    );
+
+    struct BuyInstrumentParams {
+        uint256 putStrikePrice;
+        uint256 optionAmount;
+        uint256 putMaxCost;
+        uint256 expiry;
+        uint256 lpAmt;
+        uint256 tradeAmt;
+        uint256 minWbtcAmtOut;
+        uint256 minDiggAmtOut;
+    }
+
+    using AmmAdapter for IAmmAdapter;
+    using ProtocolAdapter for IProtocolAdapter;
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    IRibbonFactory public immutable factory;
+    IProtocolAdapter public immutable adapter;
+    IAmmAdapter public immutable iUniswapAdapter;
+    IHegicOptions public immutable options;
+    AggregatorV3Interface public immutable priceProvider;
+
+    address payable public uniswapAdapterAddress;
+    string private constant adapterName = "HEGIC";
+    string private constant instrumentName = "wbtc/digg-staked-put";
+    address private constant _WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    uint256 private constant timePeriod = 2419199;
+    string private constant venue = "HEGIC";
+    uint8 private constant venueID = 1;
+
+    ProtocolAdapterTypes.OptionType public constant optionType =
+        ProtocolAdapterTypes.OptionType.Put;
+    address public immutable ethAddress;

```

Comment:
Since we are just using 0xeee for the ethAddress anyway, we can hardcode this imo. I also prefer that we use address(0) instead because of the gas benefits.